plugins/guisupport/guisupport.cpp

```diff
@@ -486,13 +486,16 @@ void GuiSupport::updateWindowIcon(QWindow *w)
     static QVector<QPixmap> gammarayIcons;
     static const auto theme = QLatin1String("gammaray");
     if (gammarayIcons.isEmpty()) {
-        gammarayIcons << QPixmap(QLatin1String(":/gammaray/images/gammaray-inject-16.png"))
-                      << QPixmap(QLatin1String(":/gammaray/images/gammaray-inject-22.png"))
-                      << QPixmap(QLatin1String(":/gammaray/images/gammaray-inject-24.png"))
-                      << QPixmap(QLatin1String(":/gammaray/images/gammaray-inject-32.png"))
-                      << QPixmap(QLatin1String(":/gammaray/images/gammaray-inject-48.png"))
-                      << QPixmap(QLatin1String(":/gammaray/images/gammaray-inject-64.png"))
-                      << QPixmap(QLatin1String(":/gammaray/images/gammaray-inject-128.png"));
+        QIcon gammarayIcon(QLatin1String(":/gammaray/images/gammaray-inject-16.png"));
+        gammarayIcon.addFile(QLatin1String(":/gammaray/images/gammaray-inject-22.png"));
+        gammarayIcon.addFile(QLatin1String(":/gammaray/images/gammaray-inject-24.png"));
+        gammarayIcon.addFile(QLatin1String(":/gammaray/images/gammaray-inject-32.png"));
+        gammarayIcon.addFile(QLatin1String(":/gammaray/images/gammaray-inject-48.png"));
+        gammarayIcon.addFile(QLatin1String(":/gammaray/images/gammaray-inject-64.png"));
+        gammarayIcon.addFile(QLatin1String(":/gammaray/images/gammaray-inject-128.png"));
+        foreach (QSize size, gammarayIcon.availableSizes()) {
```

Comment:
I would not store the pixmaps for one reason - they are dependent of the current screen dpi and moving the window to another screen with a different dpi would break rendering.
Just keep the static QIcon, and use QIcon::pixmap(window, ...) or directly QIcon::paint() when you want to update the window icon.
Also, in the event filter, I would also listen for the screen change event so you can update the icons correctly based on the screen dpi, see https://wiki.kdab.com/display/KDABTech/HiDPI for more informations.